### PR TITLE
Problem: state streaming is not supported in old versions

### PR DESCRIPTION
**Pull request by yihuang**
Solution:
- backport the feature to older version of cosmos-sdk
- integration the basic file streamer, further processing can be done based on that

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,10 @@
 - [#693](https://github.com/crypto-org-chain/cronos/pull/693) Clear `From` field when building cosmos tx in `eth_sendTransaction`.
 - [cronos#715](https://github.com/crypto-org-chain/cronos/pull/715) Add option `iavl-disable-fastnode` to disable iavl fastnode indexing migration, and enable `iavl-cache-size` to have effect on app (backport #714).
 
+### Improvements
+
+\- [#700](https://github.com/crypto-org-chain/cronos/pull/700) Integrate file streamer to stream the states of legacy blocks.
```
Comment (yihuang) [inline CHANGELOG.md:12]:
```suggestion
\- [#717](https://github.com/crypto-org-chain/cronos/pull/717) Integrate file streamer to stream the states of legacy blocks.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,10 @@
 - [#693](https://github.com/crypto-org-chain/cronos/pull/693) Clear `From` field when building cosmos tx in `eth_sendTransaction`.
 - [cronos#715](https://github.com/crypto-org-chain/cronos/pull/715) Add option `iavl-disable-fastnode` to disable iavl fastnode indexing migration, and enable `iavl-cache-size` to have effect on app (backport #714).
 
+### Improvements
+
+\- [#717](https://github.com/crypto-org-chain/cronos/pull/717) Integrate file streamer to stream the states of legacy blocks.
```
Comment (thomas-nguy) [inline CHANGELOG.md:12]:
```suggestion
- [#717](https://github.com/crypto-org-chain/cronos/pull/717) Integrate file streamer to stream the states of legacy blocks.
```

**Review by mmsqe — APPROVED** [review]

File: cmd/cronosd/cmd/root.go [hunk 3]
```diff
@@ -146,6 +147,7 @@ func initRootCmd(rootCmd *cobra.Command, encodingConfig params.EncodingConfig) {
 func addModuleInitFlags(startCmd *cobra.Command) {
 	crisis.AddModuleInitFlags(startCmd)
 	cronos.AddModuleInitFlags(startCmd)
+	startCmd.Flags().String(cronosclient.FlagStreamers, "", "Enable streamers, only file streamer is supported right now")
```
Comment (thomas-nguy) [inline cmd/cronosd/cmd/root.go:150]: nitty picky but would it be better to support enum type there?

maybe ok for the backporting version

Comment (yihuang) [reply cmd/cronosd/cmd/root.go:150]: There'll be a better configuration system for streamers in another PR in cosmos-sdk.
For this version I think we just need to generate the stream files, then convert to version db with some offline script.

**Pull request closed without merging**